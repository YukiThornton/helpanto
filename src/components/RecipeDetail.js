import React from 'react';
import PropTypes from 'prop-types';

const RecipeDetail = ({visible, title, content, createdAt, lastModifiedAt}) => {
  if (visible) {
    return (
      <div className='Recipe-Details'>
        <h2>{title}</h2>
        <p style={{color: 'gray'}}>{content}</p>
      </div>
    );
  } else {
    return <div className='Recipe-Details'></div>;
  }
};

RecipeDetail.propTypes = {
  visible: PropTypes.bool.isRequired,
  title: PropTypes.string,
  content: PropTypes.string,
  createdAt: PropTypes.string,
  lastModifiedAt: PropTypes.string,
};

export default RecipeDetail;
