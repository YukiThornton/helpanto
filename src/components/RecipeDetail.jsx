import React from 'react';
import PropTypes from 'prop-types';
import isRequiredIf from 'react-proptype-conditional-require';

const RecipeDetail = ({
  visible,
  recipe,
}) => {
  if (visible) {
    return (
      <div className="Recipe-Details">
        <h2>{recipe.title}</h2>
        <p
          style={{
            color: 'gray',
          }}
        >
          {recipe.content}
        </p>
      </div>
    );
  }
  return <div className="Recipe-Details" />;
};

RecipeDetail.propTypes = {
  visible: PropTypes.bool.isRequired,
  recipe: isRequiredIf(
    PropTypes.shape({
      title: PropTypes.string.isRequired,
      content: PropTypes.string.isRequired,
      createdAt: PropTypes.string.isRequired,
      lastModifiedAt: PropTypes.string.isRequired,
    }),
    props => props.visible,
  ),
};

export default RecipeDetail;
