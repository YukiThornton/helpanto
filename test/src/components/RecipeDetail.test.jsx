import React from 'react';
import { shallow } from 'enzyme';
import RecipeDetail from '../../../src/components/RecipeDetail';

describe('test RecipeDetail', () => {
  it('should render nothing when visible === false', () => {
    const component = (
      <RecipeDetail
        visible={false}
      />
    );
    expect(shallow(component)).toMatchSnapshot();
  });

  it('should render recipe detals when visible === true', () => {
    const component = (
      <RecipeDetail
        visible
        recipe={{
          title: 'title1',
          content: 'content1',
          createdAt: '2017-12-01T06:44:04.262Z',
          lastModifiedAt: '2017-12-01T06:44:04.262Z',
        }}
      />
    );
    expect(shallow(component)).toMatchSnapshot();
  });
});
