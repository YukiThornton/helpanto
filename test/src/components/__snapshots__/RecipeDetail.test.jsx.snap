// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test RecipeDetail should render nothing when visible === false 1`] = `
<div
  className="Recipe-Details"
/>
`;

exports[`test RecipeDetail should render recipe detals when visible === true 1`] = `
<div
  className="Recipe-Details"
>
  <h2>
    title1
  </h2>
  <p
    style={
      Object {
        "color": "gray",
      }
    }
  >
    content1
  </p>
</div>
`;
